File: app/src/main/java/id/ac/ui/cs/mobileprogramming/nabilafv/rememberyourlord/model/WeatherResponse.java
package id.ac.ui.cs.mobileprogramming.nabilafv.rememberyourlord.model;

import java.util.List;

public class WeatherResponse {
    private Coordinate coord;
    private List<WeatherInfo> weather;
    private String base;
    private WeatherDetail main;
    private Wind wind;
    private Clouds clouds;
    private long dt;
    private int timezone;
    private long id;
    private String name;

    public WeatherResponse(Coordinate coord,
                           List<WeatherInfo> weather,
                           String base,
                           WeatherDetail main,
                           Wind wind,
                           Clouds clouds,
                           long dt,
                           int timezone,
                           long id,
                           String name) {
        this.coord = coord;
        this.weather = weather;
        this.base = base;
        this.main = main;
        this.wind = wind;
        this.clouds = clouds;
        this.dt = dt;
        this.timezone = timezone;
        this.id = id;
        this.name = name;
    }

    public Coordinate getCoord() {
        return this.coord;
    }

    public List<WeatherInfo> getWeather() {
        return this.weather;
    }

    public String getBase() {
        return this.base;
    }

    public WeatherDetail getMain() {
        return this.main;
    }

    public Wind getWind() {
        return this.wind;
    }

    public Clouds getClouds() {
        return this.clouds;
    }

    public long getDt() {
        return this.dt;
    }

    public int getTimezone() {
        return this.timezone;
    }

    public long getId() {
        return this.id;
    }

    public String getName() {
        return this.name;
    }
}

File: app/src/main/java/id/ac/ui/cs/mobileprogramming/nabilafv/rememberyourlord/model/WeatherResponse.kt
package id.ac.ui.cs.mobileprogramming.nabilafv.rememberyourlord.model

data class WeatherResponse(
    val coord: Coordinate,
    val weather: List<WeatherInfo?>,
    val base: String?,
    val main: WeatherDetail,
    val wind: Wind,
    val clouds: Clouds,
    val dt: Long?,
    val timezone: Int?,
    val id: Long?,
    val name: String?
)

data class Coordinate(val lon: Double?, val lat: Double?)

data class WeatherInfo(val main: String?, val description: String?, val icon: String?)

data class WeatherDetail(
    val temp: Double?, val feelsLike: Double?, val tempMin: Double?,
    val tempMax: Double?, val pressure: Double?, val humidity: Double?
)

data class Wind(val speed: Double?, val deg: Double?)

data class Clouds(val all: Double)